clustercontroller-core/src/main/java/com/yahoo/vespa/clustercontroller/core/NodeStateChangeChecker.java

```diff
@@ -281,48 +270,38 @@ private static boolean groupContainsNode(Group group, Node node) {
         return false;
     }
 
-    private Result anyNodeSetToMaintenance(ClusterState clusterState) {
-        for (NodeInfo nodeInfo : clusterInfo.getAllNodeInfo()) {
-            if (clusterState.getNodeState(nodeInfo.getNode()).getState() == State.MAINTENANCE) {
-                return Result.createDisallowed("Another node is already in maintenance:" + nodeInfo.getNodeIndex());
+    private Result checkAllNodesAreUp(ClusterState clusterState) {
+        // This method verifies both storage nodes and distributors are up (or retired).
+        // The complicated part is making a summary error message.
+
+        for (NodeInfo storageNodeInfo : clusterInfo.getStorageNodeInfo()) {
+            State wantedState = storageNodeInfo.getUserWantedState().getState();
+            if (wantedState != State.UP && wantedState != State.RETIRED) {
+                return Result.createDisallowed("Another storage node wants state " +
+                        wantedState.toString().toUpperCase() + ": " + storageNodeInfo.getNodeIndex());
             }
-            if (nodeInfo.getWantedState().getState() == State.MAINTENANCE) {
-                return Result.createDisallowed("Another node wants maintenance:" + nodeInfo.getNodeIndex());
+
+            State state = clusterState.getNodeState(storageNodeInfo.getNode()).getState();
+            if (state != State.UP && state != State.RETIRED) {
+                return Result.createDisallowed("Another storage node has state " + state.toString().toUpperCase() +
+                        ": " + storageNodeInfo.getNodeIndex());
             }
         }
-        return Result.allowSettingOfWantedState();
-    }
 
-    private int contentNodesWithAvailableNodeState(ClusterState clusterState) {
-        final int nodeCount = clusterState.getNodeCount(NodeType.STORAGE);
-        int upNodesCount = 0;
-        for (int i = 0; i < nodeCount; ++i) {
-            final Node node = new Node(NodeType.STORAGE, i);
-            final State state = clusterState.getNodeState(node).getState();
-            if (state == State.UP || state == State.RETIRED || state == State.INITIALIZING) {
-                upNodesCount++;
+        for (NodeInfo distributorNodeInfo : clusterInfo.getDistributorNodeInfo()) {
+            State wantedState = distributorNodeInfo.getUserWantedState().getState();
+            if (wantedState != State.UP && wantedState != State.RETIRED) {
```

Comment:
Side note: as far as I can recall, distributors are not set to `retired` as they do not themselves have any data, so that particular check is not likely to ever trigger. No harm in having it around, though.